### PR TITLE
Illuminance configuration improvements

### PR DESCRIPTION
**Pull request by cdjackson**
This PR includes changes to the reporting configuration class ```ZclReportingConfig```. This now supports changing the reporting change, and also the reporting min/max times.

@triller-telekom I'd welcome your review on this.

Note that there will be a subsequent PR to improve the handling of reporting configuration which is  possible with the way reporting is now handled in the latest ZSS, but I'd like to get this merged as it already covers more than one feature.

Signed-off-by: Chris Jackson <chris@cd-jackson.com>

### CODE REVIEW

**Comment by cdjackson** [comment]
I would prefer not to have to waste time refactoring to be honest. The development of these features was interlinked which is why they ended up in a single branch and splitting them seems unnecessary even if in an ideal world we should only have one change per branch.

Is that ok? Otherwise it will be next week before I can look at this I think.

**Comment by cdjackson** [comment]
Updated.

File: org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/config/ZclReportingConfig.java [hunk 1]
```diff
@@ -66,14 +67,30 @@
     private BigDecimal minimumChange;
     private BigDecimal maximumChange;
 
+    public ZclReportingConfig(Channel channel) {
+        Configuration configuration = channel.getConfiguration();
+        if (configuration.containsKey(CONFIG_REPORTINGMIN)) {
+            reportingTimeMin = ((BigDecimal) configuration.get(CONFIG_REPORTINGMIN)).intValue();
+        }
+        if (configuration.containsKey(CONFIG_REPORTINGMIN)) {
```
Comment (triller-telekom) [inline org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/config/ZclReportingConfig.java]: I think there is a copy&paste error here: should be `CONFIG_REPORTINGMAX`

File: org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/config/ZclReportingConfig.java [hunk 2]
```diff
@@ -66,14 +67,30 @@
...
+            reportingTimeMin = ((BigDecimal) configuration.get(CONFIG_REPORTINGMAX)).intValue();
+        }
+        if (configuration.containsKey(CONFIG_REPORTINGMIN)) {
+            reportingTimeMin = ((BigDecimal) configuration.get(CONFIG_REPORTINGCHANGE)).intValue();
```
Comment (triller-telekom) [inline org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/config/ZclReportingConfig.java]: I think there is a copy&paste error here: should be `CONFIG_REPORTINGCHANGE`

**Comment by cdjackson** [comment]
> Then, this PR contains the illuminance fix and a preparation of the channelconverter to support reporting intervals as a channel configuration, I am fine with that.

No - I split the illuminance fix into #469 

I'll update with the errors - well spotted, thanks.

**Comment by cdjackson** [comment]
Thanks Stefan

**Pull request merged**